src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -175,7 +178,42 @@ internal string CreateCacheKey(CachedVaryBy varyBy)
                             .Append(value);
                     }
                 }
-                // TODO: Parse querystring params
+
+                // Vary by query params
+                if (varyBy?.Params.Count > 0)
+                {
+                    if (varyBy.Params.Count == 1 && string.Equals(varyBy.Params[0], "*", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Vary by all available query params
+                        foreach (var query in _httpContext.Request.Query.OrderBy(q => q.Key))
```

Comment:
A few issues here. 
1) order by case sensitivity? The QueryCollection operates on case-insensitive keys so maybe we need to order by `IgnoreCase` and convert all query keys to upper case. Sign perf...
2) Linq vs Array.Sort(). Was using Linq to guarantee a stable sort but I think the keys are unique so it doesn't matter? Also, what is the impact on performance?
3) Could consider using the raw `QueryString` but that approach will forgo order-insensitivity of the params. I don't think we should do this unless perf is a big problem here.